### ISSUE TITLE
a subtle bug in st_boundary.cpp

### ISSUE THREAD

**Issue by zhuzd12** [ApolloAuto/apollo#9460]
# bug description
The GetIndexRange function called in GetUnblockSRange function in st_boundary.cpp will set both index left and right zero when input curr_time equals zero, and the denominator of the formula of r equals zero consequently, resulting in NAN and incorrect range calculation in speed_boundary_decider.cpp. A similar mistake is made in GetBoundarySRange().

## how GetIndexRange() calculate the index left and right
```c++
bool STBoundary::GetIndexRange(const std::vector<STPoint>& points,
                               const double t, size_t* left,
                               size_t* right) const {
  ...
  auto comp = [](const STPoint& p, const double t) { return p.t() < t; };
  auto first_ge = std::lower_bound(points.begin(), points.end(), t, comp);
  size_t index = std::distance(points.begin(), first_ge);
  if (index == 0) {
    *left = *right = 0;
  } else if (first_ge == points.end()) {
    *left = *right = points.size() - 1;
  } else {
    *left = index - 1;
    *right = index;
  }
```

## GetUnblockSRange() calls GetIndexRange()
```c++
bool STBoundary::GetUnblockSRange(const double curr_time, double* s_upper,
                                  double* s_lower) const {
  ...
  size_t left = 0;
  size_t right = 0;
  if (!GetIndexRange(lower_points_, curr_time, &left, &right)) {
    AERROR << "Fail to get index range.";
    return false;
  }
  const double r = (curr_time - upper_points_[left].t()) /
                   (upper_points_.at(right).t() - upper_points_.at(left).t());

  double upper_cross_s =
      upper_points_[left].s() +
      r * (upper_points_[right].s() - upper_points_[left].s());
  double lower_cross_s =
      lower_points_[left].s() +
      r * (lower_points_[right].s() - lower_points_[left].s());
```

## a simple way to fix it 
```c++
   const double r = left == right ? 0.0 : (curr_time - upper_points_[left].t()) /
                   (upper_points_.at(right).t() - upper_points_.at(left).t());
```

**Comment by zhuzd12** [comment]
I wonder if anyone gives attention to this issue.

**Comment by jmtao** [comment]
@zhuzd12 Good catch! this will be fixed via PR #9763 
Thanks for supporting Apollo!
